core/src/main/java/org/apache/iceberg/BaseTransaction.java

```diff
@@ -241,7 +234,14 @@ private void commitCreateTransaction() {
     }
   }
 
-  private void commitReplaceTransaction() {
+  private void commitReplaceTransaction(boolean orCreate) {
+    if (base == null && !orCreate) {
+      throw new NoSuchTableException("Table doesn't exist");
```

Comment:
Since this isn't called inside the commit retry loop, it will always depend on the initial state of the transaction, right? In that case, the problem should be caught when the transaction is created instead.

Reply:
Yeah, it is actually handled there as well. This should never happen.

Reply:
Then I think we can remove this if/elseif/else block.